apps/frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -1,20 +1,31 @@
-import { type ReactNode } from "react";
 import { Navigate } from "react-router-dom";
 
 import { Loader } from "~/libs/components/components.js";
-import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
+import {
+	AppRoute,
+	DataStatus,
+	type PermissionKey,
+} from "~/libs/enums/enums.js";
+import { checkHasPermission } from "~/libs/helpers/helpers.js";
 import { useAppSelector } from "~/libs/hooks/hooks.js";
+import { type ValueOf } from "~/libs/types/types.js";
+import { NotFound } from "~/pages/not-found/not-found.jsx";
 
 import styles from "./styles.module.css";
 
 type Properties = {
-	children: ReactNode;
+	children: React.ReactNode;
+	permissionKeys?: ValueOf<typeof PermissionKey>[];
 };
 
-const ProtectedRoute = ({ children }: Properties): JSX.Element => {
+const ProtectedRoute = ({
+	children,
+	permissionKeys = [],
```

Comment:
let's name it by role
```suggestion
	routePermissions = [],
```